pkg/controller/controller.go

```diff
@@ -186,21 +211,27 @@ func (dc *DefaultController) run(stopChan <-chan bool) {
 
 	defer close(dc.watcher)
 
-	// start workers
-	for i := 0; i < dc.options.Workers; i++ {
-		go func() {
-			defer trace.Trace("")()
-
-			for entity := range dc.watcher {
-				func() {
-					log.Printf("received event=%s entity=%s", entity.GetStatus(), entity.GetName())
-					if err := dc.processItem(entity); err != nil {
-						log.Error(err)
-					}
-				}()
+	// Start a worker pool.  The pool scales up to dc.options.Workers.
+	go func() {
```

Comment:
this only spawns one goroutine, how does it scale to `dc.options.Workers`?

Reply:
Doh!  forgot one little key word

Reply:
Magic of Go, instead of few imports and multiple declarations, now you only need to miss two characters to make your application accidentally single-threaded ;)